src/main/java/seedu/address/ui/UserProfilePanel.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.StackPane;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.Logic;
+import seedu.address.model.person.User;
+
+/**
+ * Controller for user profile page
+ */
+public class UserProfilePanel extends UiPart<Region> {
+
+    private static final String FXML = "UserProfilePanel.fxml";
+    private static final Logger logger = LogsCenter.getLogger(UserProfilePanel.class);
+
+    private static final String PHONE_NUMBER = "Phone number: ";
+    private static final String EMAIL = "Email: ";
+    private static final String ADDRESS = "Address: ";
+    private static final String TELEGRAM = "Telegram";
+    private static final String GROUP_TAGS = "Groups: ";
+    private static final String MODULE_TAGS = "Modules: ";
+
+    private Logic logic;
+    private User user;
+
+    @FXML
+    private StackPane userProfilePane;
+    @FXML
+    private Label name;
+    @FXML
+    private Label phone;
+    @FXML
+    private Label email;
+    @FXML
+    private Label address;
+    @FXML
+    private Label telegramHandle;
+    @FXML //TODO: MIGHT CHANGE TO FLOW PANE LATER ON
+    private Label groupTags;
+    @FXML //TODO: MIGHT CHANGE TO FLOW PANE LATER ON
+    private Label moduleTags;
+
+    /**
+     * Creates a new User profile window.
+     */
+    public UserProfilePanel(Logic logic) {
+        super(FXML);
+        this.logic = logic;
+        this.user = this.logic.getUser();
+
+        name.setText(String.valueOf(this.user.getName()));
```

Comment:
I guess can use this.user.getName().toString() or "" + this.user.getName(), the other values don't need it because it has a string appended at the front.